core/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunctionParser.java

```diff
@@ -22,23 +22,18 @@
 import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.ParsingException;
 import org.elasticsearch.common.bytes.BytesReference;
-import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.index.query.QueryParseContext;
-import org.elasticsearch.index.query.functionscore.gauss.GaussDecayFunctionBuilder;
-import org.elasticsearch.index.query.functionscore.gauss.GaussDecayFunctionParser;
 import org.elasticsearch.search.MultiValueMode;
 
 import java.io.IOException;
+import java.util.function.BiFunction;
 
 /**
- * This class provides the basic functionality needed for adding a decay
- * function.
- *
- * This parser parses this kind of input
+ * Superclass for all parsers for decay functions. It parses this kind of input:
```

Comment:
nit: can this still be called a superclass given that it has no subclasses? Maybe it should be final actually?